components/content-search/index.tsx

```diff
@@ -514,20 +330,29 @@ const ContentSearch: React.FC<ContentSearchProps> = ({
 								})}
 						</List>
 
-						{!isLoading && hasSearchResults && showLoadMore && (
+						{hasSearchResults && hasNextPage && (
 							<LoadingContainer>
-								<Button onClick={handleLoadMore} variant="secondary">
+								<Button onClick={() => fetchNextPage()} variant="secondary">
 									{__('Load more', '10up-block-components')}
 								</Button>
 							</LoadingContainer>
 						)}
 
-						{isLoading && currentPage > 1 && <StyledSpinner onPointerEnterCapture={null} onPointerLeaveCapture={null} />}
+						{isFetchingNextPage && <StyledSpinner onPointerEnterCapture={null} onPointerLeaveCapture={null} />}
 					</>
 				) : null}
 			</StyledNavigableMenu>
+	);
+};
+
+const ContentSearchWrapper: React.FC<ContentSearchProps> = (props) => {
+	return (
+		<StyledComponentContext cacheKey="tenup-component-content-search">
+			<QueryClientProvider client={queryClient}>
```

Comment:
@fabiankaegy this looks sane to me. I don't think this is a huge issue as it stands. The chances of this being super needed because you have multiple modules being used by a single user seem very slim. Also the extra calls are not as important on the admin side so, as you say, it is the next best thing. 

Been looking to see if there was any way it could share some context but it does not seem possible (though it used to be).